### ISSUE TITLE
Base controller

### ISSUE THREAD

**Issue by ababakov** [Romakita/ts-express-decorators#91]
## Informations

Type | Version
---|---
Question | 2.x

****

## Description

Is it possible to use a generic base controller for common crud operations?
```typescript
class BaseController<T> {

  constructor(
    protected entriesService: BaseEntryService<T>
  ) {
  }

  @Get('/list')
  index(): any {
    return {list: []};
  }
}

@Controller('/users')
export default class UsersCtrl extends BaseController<IUser> {
  constructor(
    private usersService: UsersService
  ) {
    super(usersService);
  }
  ...
}
```

I see '/api/users/list' in the routes table

 GET              │ /api/users/list                 │ UsersCtrl.index() 

But getting

GET /api/users/list 404 1.662 ms - 146

**Comment by Romakita** [comment]
Hello @ababakov,

It's a strange behavior... Regarding to this subject https://stackoverflow.com/questions/36837421/extending-component-decorator-with-base-class-decorator/36837482  decorator can't used with a parent class.

So I don't know how the route resolver have found this configuration ^^

The right way to create dynamic routes  (not the best), is to inject the [RouterController](https://romakita.github.io/ts-express-decorators/#/api/common/mvc/routercontroller) service in your controller and use the `Express.Router` to add the routes (without decorators). But in this case, all routes added by RouterController will not appear on the routes table.

Example:
```typescript
class BaseController<T> {
  private router;
  constructor(
    routerController: RouterController,
    protected entriesService: BaseEntryService<T>
  )  {
      this.router  = this.routerController.getRouter()
      this.router.get('/', this.index)
  }

  index = (): any => {
    return {list: []};
  }
}

@Controller('/users')
export default class UsersCtrl extends BaseController<IUser> {
  constructor(
    routerController: RouterController,
    private usersService: UsersService
  ) {
    super(routerController, usersService);
  }
  ...
}
```

@WalterLeinert I think you have already worked on the same problem? have you find a solution?

See you,
Romain

**Comment by Romakita** [comment]
Hi @ababakov,
 I work on a fix to allow inheritance from a BaseController :)

https://github.com/Romakita/ts-express-decorators/tree/fix_extending_controllers

See you,
Romain

**Comment by Romakita** [comment]
https://github.com/Romakita/ts-express-decorators/pull/93

**Comment by Romakita** [comment]
Fix with v2.0.12
